### PR TITLE
[JENKINS-32196] Added support for the Maven Info Plugin

### PR DESCRIPTION
**Pull request by v1v**
Plugin Wiki: https://wiki.jenkins-ci.org/display/JENKINS/Maven+Info+Plugin#MavenInfoPlugin-AssignnamefromPOM
Jira ticket: https://issues.jenkins-ci.org/browse/JENKINS-32196

### CODE REVIEW

File: job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/properties/MavenPropertiesContext/mavenInfo.groovy [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        }
+    }
+}
+
+mavenJob('example_2') {
+    properties {
+        mavenInfo {
+            assignDescription {
+                descriptionTemplate('some text')
+                modulePattern(':my-artifact')
```
Comment (daspilker) [inline job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/helpers/properties/MavenPropertiesContext/mavenInfo.groovy]: Aren't the module and dependencies patterns independent from the description option? You can click the "Advanced..." button even if "Assign Description" is not selected. So the DSL should be like

```groovy
mavenInfo {
    assignName('name template')
    assignDescription('some text')
    modulePattern(':my-artifact')
    interestingDependenciesPattern('org.springframework.*:*')
}
```

Those settings are also used with "assignName", see https://github.com/jenkinsci/maven-info-plugin/blob/master/src/main/java/jenkins/plugins/maveninfo/extractor/properties/NamePropertiesFinder.java#L21-L22 for an example.

**Comment by daspilker** [comment]
Do you want to pick this up? Otherwise I will close the PR.

**Comment by v1v** [comment]
I'll review it shortly

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/properties/MavenInfoContext.groovy [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    void assignName(String name) {
+        this.assignName = true
+        this.name = name
+    }
+
+    /**
+     * Sets the description from main module pom.xml (from tag <description>).
+     */
+    void assignDescription(@DslContext(MavenInfoDescriptionContext) Closure closure = null) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/properties/MavenInfoContext.groovy]: this does not need a nested context, use the same pattern as for `assignName`

**Comment by daspilker** [comment]
Please remove the `MavenInfoDescriptionContext`, rebase to master and squash your commits.

**Pull request merged**